src/manifestoo/commands/list_depends.py

```diff
@@ -11,16 +13,27 @@ def list_depends_command(
     addons_set: AddonsSet,
     transitive: bool = False,
     include_selected: bool = False,
+    sort_type: str = "alphabetical",
 ) -> Tuple[Iterable[str], Iterable[str]]:
-    result: Set[str] = set()
+    result_dict: Dict[str, Set[str]] = {}
     missing: Set[str] = set()
     for addon_name, addon in dependency_iterator(
         addons_selection, addons_set, transitive
     ):
         if include_selected:
-            result.add(addon_name)
+            result_dict[addon_name] = set(addon.manifest.depends) if addon else set()
         if not addon:
             missing.add(addon_name)
         else:
-            result.update(set(addon.manifest.depends) - set(addons_selection))
-    return sorted(result), missing
+            depends_to_add = set(addon.manifest.depends) - set(addons_selection)
+            for depend_to_add in depends_to_add:
+                addon_to_add = addons_set.get(depend_to_add)
+                result_dict[depend_to_add] = (
+                    set(addon_to_add.manifest.depends) if addon_to_add else set()
+                )
+    if sort_type == "topological":
+        topological_sorted_res = TopologicalSorter(result_dict)
+        result: List[str] = list(topological_sorted_res.static_order())
+    else:
+        result = sorted(result_dict.keys())
+    return result, missing
```

Comment:
Since this will be used for the `list` command too, let's create a sorting method that accepts a sequence of addon names and an AddonSet and returns the sorted addon names. This could take the form of an AddonSorter abstract class with two concrete implementations. And the list_depends_command function can then accept an AddonSorter instance to perform the sorting.